packages/block-editor/src/components/plain-text/index.native.js

```diff
@@ -79,20 +82,74 @@ export default class PlainText extends Component {
 		};
 	}
 
+	replaceLineBreakTags( value ) {
+		return value?.replace( RegExp( '<br>', 'gim' ), '\n' );
+	}
+
+	onChangeTextInput( event ) {
+		const { onChange } = this.props;
+		onChange( event.nativeEvent.text );
+	}
+
+	onChangeRichText( value ) {
+		const { onChange } = this.props;
+		// The <br> tags have to be replaced with new line characters
+		// as the content of plain text shouldn't contain HTML tags.
+		onChange( this.replaceLineBreakTags( value ) );
+	}
+
 	render() {
-		const { style } = this.props;
+		const {
+			style,
+			__experimentalVersion,
+			onFocus,
+			...otherProps
+		} = this.props;
 		const textStyles = [
 			style || styles[ 'block-editor-plain-text' ],
 			this.getFontSize(),
 		];
 
+		if ( __experimentalVersion === 2 ) {
+			const disableFormattingProps = {
+				withoutInteractiveFormatting: true,
+				disableEditingMenu: true,
+				__unstableDisableFormats: true,
+				disableSuggestions: true,
+			};
+
+			const forcePlainTextProps = {
+				preserveWhiteSpace: true,
+				__unstablePastePlainText: true,
+				multiline: false,
+			};
+
+			const fontProps = {
+				fontFamily: style?.fontFamily,
+				fontSize: style?.fontSize,
+				fontWeight: style?.fontWeight,
+			};
+
+			return (
+				<RichText
+					{ ...otherProps }
+					{ ...disableFormattingProps }
+					{ ...forcePlainTextProps }
+					{ ...fontProps }
+					identifier="content"
+					tagName="pre"
```

Comment:
I think this should come from props, currently, it's fine because only the `Shortcode` block uses it but if we want to use it in other blocks, the `tagName` would be different, what do you think?

Reply:
Yep, good point. In fact, I think that we could even remove the prop to use the default value `div` ([reference](https://github.com/WordPress/gutenberg/blob/51db4bf888e6b18cf9d18266108114d61070f3ad/packages/rich-text/src/component/index.native.js#L1248)). As far as I checked, it works even better than using `pre` tag 😅 .

Android|iOS
--|--
<video src=https://user-images.githubusercontent.com/14905380/169492308-52798d75-c6d2-4f44-8a42-270e2745924a.mp4>|<video src=https://user-images.githubusercontent.com/14905380/169492332-9e9582ae-0840-4e3e-a86d-95a01a155505.mp4>

As an additional note, I'd like to share that during implementation, I tried using different tag names like `p` or `span`  but found some issues:
- `p`: On Android, it doesn't allow to add new lines at the end of the content.
https://github.com/WordPress/gutenberg/blob/51db4bf888e6b18cf9d18266108114d61070f3ad/packages/rich-text/src/component/index.native.js#L1095-L1097
- `span`: Automatically wraps the `span` tags with `p` tags. Besides, adding new lines leads to odd issues where lines are merged or add extra newlines 🙃.

Reply:
Sounds good! At first I was wondering how it was showing the monospace font without the `pre` tag but I checked we are setting the font from the styles so we are good 👍